esphome/components/tem3200/tem3200.cpp

```diff
@@ -0,0 +1,154 @@
+#include "tem3200.h"
+#include "esphome/core/log.h"
+#include "esphome/core/helpers.h"
+#include "esphome/core/hal.h"
+
+namespace esphome {
+namespace tem3200 {
+
+static const char *const TAG = "tem3200";
+
+enum ErrorCode {
+  NONE = 0,
+  RESERVED = 1,
+  STALE = 2,
+  FAULT = 3,
+};
+
+void TEM3200Component::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up TEM3200...");
+
+  uint8_t status(NONE);
+  uint16_t temperature_raw(0);
+  uint16_t pressure_raw(0);
+
+  // startup device delay
+  delay(10);
+
+  i2c::ErrorCode err = this->read_(status, temperature_raw, pressure_raw);
```

Comment:
Should a read error during setup have a retry or two?

Reply:
That is a good idea...